lib/bitex_bot/models/closing_flow.rb

```diff
@@ -5,157 +5,82 @@ class ClosingFlow < ActiveRecord::Base
+    # Start a new CloseBuy that closes existing OpenBuy's by selling on taker market what was just bought on maker market.
     # rubocop:disable Metrics/AbcSize
-    def self.close_open_positions
-      return unless open_positions.any?
+    def self.close_market
+      return unless open_position_class.open.any?
 
-      positions = open_positions
+      positions = open_position_class.open
       quantity = positions.sum(&:quantity)
-      amount = positions.sum(&:amount) / fx_rate
-      price = suggested_amount(positions) / quantity
-      unless Robot.taker.enough_order_size?(quantity, price)
-        Robot.log(
-          :info,
-          "Closing: #{Robot.taker.name} - enough order size for #{Robot.taker.base.upcase} #{quantity}"\
-          " @ #{Robot.taker.quote.upcase} #{price}"
-        )
-        return
-      end
-
-      create_closing_flow!(price, quantity, amount, positions)
-    end
-    # rubocop:enable Metrics/AbcSize
-
-    def self.open_positions
-      open_position_class.open
-    end
+      price = (suggested_amount(positions) / quantity)
+      return unless Robot.taker.enough_order_size?(quantity, price)
 
-    # close_open_positions helpers
-    def self.suggested_amount(positions)
-      positions.map { |p| p.quantity * p.opening_flow.suggested_closing_price }.sum
-    end
+      order = Robot.taker.place_order(trade_type, price, quantity)
+      Robot.log(:info, "Closing: placed #{trade_type} with price: #{order.price} @ quantity #{order.amount}.")
 
-    def self.create_closing_flow!(price, quantity, amount, open_positions)
-      flow = create!(desired_price: price, quantity: quantity, amount: amount, open_positions: open_positions)
-      Robot.log(
-        :info,
-        "Closing: created #{self}##{flow.id}, desired price: #{flow.desired_price}, quantity: #{flow.quantity}, "\
-        "amount: #{flow.amount}."
-      )
-      flow.create_initial_order_and_close_position!
-      nil
-    end
-    # end: close_open_positions helpers
+      amount = positions.sum(&:amount) / fx_rate
 
-    def create_initial_order_and_close_position!
-      create_order_and_close_position(quantity, desired_price)
+      create!(desired_price: price, quantity: quantity, amount: amount, open_positions: positions).tap do |flow|
+        flow.close_positions.create!(order_id: order.id)
+      end
+    rescue StandardError => e
+      raise CannotCreateFlow, e.message
     end
+    # rubocop:enable Metrics/AbcSize
 
-    # TODO: should receive a order_ids and user_transaccions array, then each Wrapper should know how to search for them.
-    def sync_closed_positions
-      # Maybe we couldn't create the bitstamp order when this flow was created, so we try again when syncing.
-      latest_close.nil? ? create_initial_order_and_close_position! : create_or_cancel!
-    end
+    # In this steps if exist anyone active closing flow, asume that also will has a anyone close position.
```

Comment:
![tevez-1-569x420](https://user-images.githubusercontent.com/130087/54356651-e2491a80-463a-11e9-9fda-34ebb5d9c87c.jpg)

Reply:
At this point, if any active closing flows exist, it also means there are close positions created.